packages/flutter_tools/test/commands.shard/permeable/build_bundle_test.dart

```diff
@@ -536,16 +543,77 @@ void main() {
     await runner.run(<String>[
       'bundle',
       '--no-pub',
-      '--dart-define-from-file=config.json',
+      '--dart-define-from-file=config1.json',
+      '--dart-define-from-file=config2.json',
     ]);
   }, overrides: <Type, Generator>{
-    BuildSystem: () => TestBuildSystem.all(BuildResult(success: true), (Target target, Environment environment) {
-      expect(environment.defines[kDartDefines], 'a0ludD0x,a0RvdWJsZT0xLjE=,bmFtZT1kZW5naGFpemh1,dGl0bGU9dGhpcyBpcyB0aXRsZSBmcm9tIGNvbmZpZyBqc29uIGZpbGU=');
+    BuildSystem: () => TestBuildSystem.all(BuildResult(success: false), (Target target, Environment environment) {
+      expect(environment.defines[kDartDefines], 'a0ludD0x,a0RvdWJsZT0xLjE=,bmFtZT1kZW5naGFpemh1,dGl0bGU9dGhpcyBpcyB0aXRsZSBmcm9tIGNvbmZpZyBqc29uIGZpbGU=,Ym9keT10aGlzIGlzIGJvZHkgZnJvbSBjb25maWcganNvbiBmaWxl');
```

Comment:
This is my first time seeing this test--what encoding is this?

Reply:
Also, are you intentionally now testing that this fails?

Reply:
These are encoded from dartDefines' key and value on `encodeDartDefines` in build_info.dart.
For example, kInt=1 is encoded to a0ludD0x.

I intended this test should success, this is my mistake. But, I found that assertions on this test file doesn't work as expected because of `FakeBundleBuilder` that do no-op on build function.
I'll investigate more and fix them to work correctly.

Reply:
I fixed to use real `BundleBuilder` so that we can inject BuildSystem on `build_bundle_test.dart` and assert environment properties.
I checked tests work as expected and also fix `flutter_command.dart`.

Reply:
Ahh, so base64. I'm skeptical this is the best way to assert here (given I had to decode it to code review), however, that can be a cleanup in a later PR.